src/topology/sheaves/stalks.lean

```diff
@@ -174,4 +219,175 @@ begin
 end
 
 end stalk_pushforward
+
+variables {C}
+
+/--
+The induced map of a morphism `f : F ⟶ G` of presheaves on the stalks
+-/
+def stalk_map (x : X) {F G : X.presheaf C} (f : F ⟶ G): F.stalk x ⟶ G.stalk x :=
+(stalk_functor C x).map f
```

Comment:
My instinct is that life would be easier if you just worked in terms of `stalk_functor`, rather than introducing an intermediate definition. Similarly for `iso` below. I'm not particularly certain of this, however.

Reply:
You might be right, I will try it and see how it goes. In particular I was disappointed that the simp lemmas `germ_comp` and `germ_comp_apply` never seem to fire automatically, I always needed to rewrite with them explicitly. Maybe removing an extra layer of definitions will help with that?
Or what would your instinct say are the criterions to look for here?

Reply:
Yes, `germ_comp` won't ever work if the expression involves `stalk_functor`, rather than `stalk_map`.

You could try making `stalk_map` an `abbreviation` rather than a `def`, but give it's not saving many characters it may be easier to just leave it out entirely.

Reply:
I guess the main difference is that for `stalk_map`, the category `C` is implicit while for `stalk_functor`, it is explicit. So it might save more characters, e.g. `stalk_map x f` instead of `(stalk_functor (Type v) x).map f`. I can definitely change it to an `abbreviation` though. But if you say we should leave it out, I'm fine with that too. I'm still very new to library building in Lean :)

Reply:
Okay, I removed the `stalk_map` definition now and I think it looks way better: 282d8da966b31d84cc8cf6f01c0ea3edf60825a0